tests/datalab/test_issue_manager.py

```diff
@@ -68,3 +73,25 @@ def find_issues(self):
             for text in ["Warning: Overwriting existing issue manager foo with ", "NewFoo"]
         ]
     ), "Should print a warning"
+
+    # Reregistering for task should overwrite the existing class, put print a warning
+    class NewerFoo(IssueManager):
+        issue_name = "label"
+
+        def find_issues(self):
+            pass
+
+    NewerFoo = register(NewerFoo, task="classification")
+
+    assert "label" in REGISTRY
+    assert REGISTRY["label"] == NewerFoo
+    assert all(
+        [
+            text in sys.stdout.getvalue()
+            for text in [
+                "Warning: Overwriting existing issue manager label with ",
+                "NewerFoo",
+                " for task classification.",
+            ]
+        ]
+    ), "Should print a warning"
```

Comment:
For consistency in the test, let the `NeweFoo.issue_name = "foo"`.

- Wouldn't this issue manager be found in the `TASK_SPECIFIC_REGISTRY`?
- The way the test is written, I understand that it wouldn't overwrite an existing issue manager . I recommend you use `@pytest.mark.parametrize` to test all possible values of `task`.